extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/JsonWebKeySet.java

```diff
@@ -43,12 +47,16 @@ private void initKeys(String json) {
                     if (x5t != null && jwkKey.getKey() != null) {
                         keysWithThumbprints.put(x5t, jwkKey.getKey());
                     }
+                    if (jwkKey.getKeyId() == null && x5t == null && jwkKey.getKeyType() != null) {
+                        List<Key> keys = keysWithoutKeyIdAndThumbprint.get(jwkKey.getKeyType());
+                        if (keys == null) {
+                            keys = new LinkedList<>();
```

Comment:
Minor gripe: Why a `LinkedList`?

Reply:
Yikes, prefer an `ArrayList` instead, see https://stackoverflow.com/questions/322715/when-to-use-linkedlist-over-arraylist-in-java

Reply:
Sure, I tend to use when I don't know the final size, but looks like it is not performant at all, so let me change it :-), thanks

Reply:
Sanne also mentioned about this in: https://github.com/quarkusio/quarkus/pull/18355#discussion_r666155865

Reply:
Thanks @gastaldi